### PR TITLE
added metadata to flags.applied.npz

### PR DESCRIPTION
**Pull request by jsdillon**
Addresses #148

### CODE REVIEW

**Comment by adampbeardsley** [comment]
Don't we want the full arrays (e.g. time, lst, baseline)? Otherwise you can't immediately use the full flag_array.

File: hera_qm/xrfi.py [hunk 1]
```diff
@@ -735,7 +735,14 @@ def xrfi_apply(filename, args, history):
     else:
         raise ValueError('Unrecognized output file format ' + str(args.outfile_format))
     if args.output_npz:
-        # Save an npz with the final flag array and waterfall
+        # Save an npz with the final flag array and waterfall and relevant metadata
         outpath = outpath + args.out_npz_ext
-        np.savez(outpath, flag_array=uvd.flag_array, waterfall=wf_full,
-                 history=flag_history + history)
+        lsts = []
+        for l in uvd.lst_array.ravel():
```
Comment (zakiali) [inline hera_qm/xrfi.py]: does np.unique not work for uvd.lst_array?

Comment (nkern) [reply hera_qm/xrfi.py]: nope... that reorders it when it could be wrapped inherently. think about an lstarray that is [4, 4.5, 5, 5.5, 6, 0.25, ...]

**Comment by jsdillon** [comment]
This is all just copy-pasted from hera_cal.io.load_vis so that I get all the same outputs (except the data of course).

**Comment by jsdillon** [comment]
@adampbeardsley This is all just copy-pasted from hera_cal.io.load_vis so that I get all the same outputs (except the data of course)... this assumes that all baselines are there for all times... as does much of hera_cal.

**Comment by adampbeardsley** [comment]
Also, should we be adding this to the other npzs that are saved in xrfi_run?

**Comment by jsdillon** [comment]
Maybe... but this is what I need right now. 

Let's come back later to revisit the xrfi data products when we decide how to reorganize.

**Comment by adampbeardsley** [comment]
@jsdillon even if you assume the baselines are all there, do you get the order you need?

**Comment by jsdillon** [comment]
Good point... let me rethink this.

**Comment by jsdillon** [comment]
I forgot that you can't still call .get_flags(). So now I'm doing as you suggest, @adampbeardsley. Thoughts?

**Comment by jsdillon** [comment]
And thanks for a quick turnaround and finding my error.

**Pull request merged**